iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -457,17 +462,20 @@ public List<ColumnStatisticsObj> getColStatistics(org.apache.hadoop.hive.ql.meta
     Table table = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
     String statsPath = getStatsPath(table).toString();
     LOG.info("Using stats from puffin file at: {}", statsPath);
+    return readColStats(table, statsPath).equals(null) ? null : readColStats(table, statsPath).getStatsObj();
```

Comment:
what is this equals null check?  why are you calling `readColStats` twice??

Reply:
fixed